
C:\src\dev\wxWindows>cvs -d:pserver:anoncvs@cvs.wxwindows.org:/home/wxcvs -z7 diff -u2 -w       
? mdi.patch
? log.patch
? lib/wx22_7.dll
? lib/wx22_7.exp
? lib/wx22_7d.ilk
? lib/wx22_7d.dll
? lib/wx22_7d.exp
? samples/sockets/test.html
? samples/sockets/test.url
? src/common/lex_yy.c
Index: include/wx/log.h
===================================================================
RCS file: /home/wxcvs/wxWindows/include/wx/log.h,v
retrieving revision 1.52.2.4
diff -u -2 -w -r1.52.2.4 log.h
--- include/wx/log.h	2000/07/01 12:01:17	1.52.2.4
+++ include/wx/log.h	2001/06/02 10:45:56
@@ -104,4 +104,6 @@
     wxLog();
 
+	static void CleanupOnExit();
+	
     // these functions allow to completely disable all log messages
         // is logging disabled now?
Index: src/common/log.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/src/common/log.cpp,v
retrieving revision 1.93.2.3
diff -u -2 -w -r1.93.2.3 log.cpp
--- src/common/log.cpp	2001/04/21 01:34:57	1.93.2.3
+++ src/common/log.cpp	2001/06/02 10:46:00
@@ -474,4 +474,9 @@
 wxArrayString   wxLog::ms_aTraceMasks;
 
+void wxLog::CleanupOnExit()
+{
+    ms_aTraceMasks.Clear();
+}
+
 // ----------------------------------------------------------------------------
 // stdout error logging helper
Index: src/gtk/app.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/src/gtk/app.cpp,v
retrieving revision 1.124.2.11
diff -u -2 -w -r1.124.2.11 app.cpp
--- src/gtk/app.cpp	2001/03/08 07:57:52	1.124.2.11
+++ src/gtk/app.cpp	2001/06/02 10:46:00
@@ -594,4 +594,9 @@
     wxClassInfo::CleanUpClasses();
 
+#if wxUSE_LOG
+    // Free global logging resources right before we dump memory leaks.
+    wxLog::CleanupOnExit();
+#endif // wxUSE_LOG
+
     // check for memory leaks
 #if (defined(__WXDEBUG__) && wxUSE_MEMORY_TRACING) || wxUSE_DEBUG_CONTEXT
Index: src/mac/app.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/src/mac/app.cpp,v
retrieving revision 1.12.2.1
diff -u -2 -w -r1.12.2.1 app.cpp
--- src/mac/app.cpp	2001/03/08 07:57:52	1.12.2.1
+++ src/mac/app.cpp	2001/06/02 10:46:01
@@ -474,4 +474,9 @@
   wxTheApp = NULL;
 
+#if wxUSE_LOG
+    // Free global logging resources right before we dump memory leaks.
+  wxLog::CleanupOnExit();
+#endif // wxUSE_LOG
+
 #if (defined(__WXDEBUG__) && wxUSE_MEMORY_TRACING) || wxUSE_DEBUG_CONTEXT
   // At this point we want to check if there are any memory
Index: src/motif/app.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/src/motif/app.cpp,v
retrieving revision 1.42.2.1
diff -u -2 -w -r1.42.2.1 app.cpp
--- src/motif/app.cpp	2001/03/08 07:57:52	1.42.2.1
+++ src/motif/app.cpp	2001/06/02 10:46:01
@@ -179,4 +179,9 @@
 #endif
 
+#if wxUSE_LOG
+    // Free global logging resources right before we dump memory leaks.
+    wxLog::CleanupOnExit();
+#endif // wxUSE_LOG
+
 #if (defined(__WXDEBUG__) && wxUSE_MEMORY_TRACING) || wxUSE_DEBUG_CONTEXT
     // At this point we want to check if there are any memory
Index: src/msw/app.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/src/msw/app.cpp,v
retrieving revision 1.131.2.9
diff -u -2 -w -r1.131.2.9 app.cpp
--- src/msw/app.cpp	2001/05/17 23:44:31	1.131.2.9
+++ src/msw/app.cpp	2001/06/02 10:46:01
@@ -610,4 +610,9 @@
     wxTheApp = NULL;
 
+#if wxUSE_LOG
+    // Free global logging resources right before we dump memory leaks.
+    wxLog::CleanupOnExit();
+#endif // wxUSE_LOG
+
 #if (defined(__WXDEBUG__) && wxUSE_MEMORY_TRACING) || wxUSE_DEBUG_CONTEXT
     // At this point we want to check if there are any memory
Index: src/os2/APP.CPP
===================================================================
RCS file: /home/wxcvs/wxWindows/src/os2/APP.CPP,v
retrieving revision 1.27.2.14
diff -u -2 -w -r1.27.2.14 APP.CPP
--- src/os2/APP.CPP	2000/10/02 02:35:58	1.27.2.14
+++ src/os2/APP.CPP	2001/06/02 10:46:04
@@ -383,4 +383,9 @@
     wxTheApp = NULL;
 
+#if wxUSE_LOG
+    // Free global logging resources right before we dump memory leaks.
+    wxLog::CleanupOnExit();
+#endif // wxUSE_LOG
+
 #if (defined(__WXDEBUG__) && wxUSE_MEMORY_TRACING) || wxUSE_DEBUG_CONTEXT
     // At this point we want to check if there are any memory
Index: src/stubs/app.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/src/stubs/Attic/app.cpp,v
retrieving revision 1.11
diff -u -2 -w -r1.11 app.cpp
--- src/stubs/app.cpp	1999/12/14 23:52:04	1.11
+++ src/stubs/app.cpp	2001/06/02 10:46:04
@@ -121,4 +121,9 @@
   delete wxTheApp;
   wxTheApp = NULL;
+
+#if wxUSE_LOG
+  // Free global logging resources right before we dump memory leaks.
+  wxLog::CleanupOnExit();
+#endif // wxUSE_LOG
   
 #if (defined(__WXDEBUG__) && wxUSE_MEMORY_TRACING) || wxUSE_DEBUG_CONTEXT

 	  	 
